### ISSUE TITLE
PSScriptAnalyzer problems don't refresh after changing the settings file

### ISSUE THREAD

**Issue by revolter** [PowerShell/vscode-powershell#3168]
Issue Description
=====

I am experiencing a problem with problems not refreshing after changing the settings file.

Steps to reproduce
=====

1. Open a PowerShell script with at least one problem.
2. Open the `PSScriptAnalyzerSettings.psd1` file.
3. Add the name of the rule that's causing the problem under the `ExcludeRules` key.
4. Go back to the initial script.

Current behaviour
=====

The problem doesn't disappear.

Expected behaviour
=====

The problem should disappear.

Attached Logs
=====

Follow the instructions in the [README](https://github.com/PowerShell/vscode-powershell/blob/master/docs/troubleshooting.md) about
capturing and sending logs.

[1612637680-c7543aa0-3145-4deb-9204-e328d43685271612637672916.zip](https://github.com/PowerShell/vscode-powershell/files/5937459/1612637680-c7543aa0-3145-4deb-9204-e328d43685271612637672916.zip)

Environment Information
=====

Visual Studio Code
-----

| Name | Version |
| --- | --- |
| Operating System | Darwin x64 20.3.0 |
| VSCode | 1.53.0|
| PowerShell Extension Version | 2020.6.0 |

PowerShell Information
-----

|Name|Value|
|---|---|
|PSVersion|7.1.1|
|PSEdition|Core|
|GitCommitId|7.1.1|
|OS|Darwin 20.3.0 Darwin Kernel Version 20.3.0: Thu Jan 21 00:07:06 PST 2021, root:xnu-7195.81.3~1/RELEASE_X86_64|
|Platform|Unix|
|PSCompatibleVersions|1.0 2.0 3.0 4.0 5.0 5.1.10032.0 6.0.0 6.1.0 6.2.0 7.0.0 7.1.1|
|PSRemotingProtocolVersion|2.3|
|SerializationVersion|1.1.0.1|
|WSManStackVersion|3.0|

Visual Studio Code Extensions
-----

<details><summary>Visual Studio Code Extensions(Click to Expand)</summary>

|Extension|Author|Version|
|---|---|---|
|cpptools|ms-vscode|1.2.0|
|platformio-ide|platformio|2.2.1|
|powershell|ms-vscode|2020.6.0|
|prettier-vscode|esbenp|5.9.1|
|remote-containers|ms-vscode-remote|0.158.0|
|vetur|octref|0.32.0|
|vscode-docker|ms-azuretools|1.9.1|
|vscode-typescript-tslint-plugin|ms-vscode|1.3.3|
|vsliveshare|ms-vsliveshare|1.0.3629|

**Comment by SydneyhSmith** [comment]
Thanks @revolter PSSA is run based on events in in the target file...with our current architecture it is expensive to run the analysis based on an update to the PSSA settings, however, the analysis should get triggered by any event in the target file...

**Comment by revolter** [comment]
Does selecting the opened tab with the target file count as an event?

**Comment by rjmholt** [comment]
No, the events are defined by the Language Server Protocol and bringing a file into focus isn't one of them. You can find them described [here](https://microsoft.github.io/language-server-protocol/specifications/specification-3-16/#textDocument_synchronization).
